### PR TITLE
Implement plan sync intent service

### LINKED ISSUE #162: Implement Plan sync intent service
This sync intent service pulls from https://github.com/OpenSRP/opensrp-server-web/pull/47

### PR DESCRIPTION
**Pull request by vincent-karuri**
Closes #162 .

Depends on https://github.com/OpenSRP/opensrp-client-core/pull/220.

### CODE REVIEW

File: opensrp-reveal/src/main/java/org/smartregister/reveal/util/Utils.java [hunk 1]
```diff
@@ -88,6 +89,7 @@ public static String getPropertyValue(Feature feature, String propertyKey) {
     public static void startImmediateSync() {
         RevealCampaignServiceJob.scheduleJobImmediately(RevealCampaignServiceJob.TAG);
         PullUniqueIdsServiceJob.scheduleJobImmediately(PullUniqueIdsServiceJob.TAG);
+        PlanIntentServiceJob.scheduleJobImmediately(PlanIntentServiceJob.TAG);
```
Comment (githengi) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/util/Utils.java]: Replace Campaign service with PlanIntentServiceJob

File: opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/LoginInteractor.java [hunk 2]
```diff
@@ -23,10 +25,14 @@ protected void scheduleJobsPeriodically() {
 
         PullUniqueIdsServiceJob.scheduleJob(SyncServiceJob.TAG, TimeUnit.MINUTES.toMillis(
                 BuildConfig.PULL_UNIQUE_IDS_MINUTES), getFlexValue(BuildConfig.PULL_UNIQUE_IDS_MINUTES));
+
```
Comment (githengi) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/LoginInteractor.java]: Replace Campaign service with PlanIntentServiceJob.

Remove Campaign service

**Comment by githengi** [comment]
Since plan is the main intent service chain the other jobs on the a a child Plan intent service (should inherit  core  Plan intent service)
org.smartregister.reveal.sync.RevealCampaignIntentService#onHandleIntent

File: opensrp-reveal/src/main/java/org/smartregister/reveal/util/Utils.java (hunk 1 above)
Comment (vincent-karuri) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/util/Utils.java]: Done.

File: opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/LoginInteractor.java (hunk 2 above)
Comment (vincent-karuri) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/LoginInteractor.java]: Done.

**Pull request merged**